<?xml version="1.0"?>
<launch>
    <arg name="launch_hw" default="true" />
    <arg name="sensor_model" description="TutorialSensor"/>
    <arg name="return_mode" default="LastStrongest" description="See readme for supported return modes"/>
    <arg name="frame_id"  default="tutorial"/>
    <arg name="scan_phase" default="0.0" />

    <arg name="packet_mtu_size" default="1500" description="Packet MTU size"/>
    <arg name="dual_return_distance_threshold" default="0.1" description="Distance threshold of dual return mode"/>

    <arg name="calibration_file" default=""/>

    <arg name="sensor_ip" default="192.168.1.201" description="Lidar Sensor IP"/>
    <arg name="host_ip" default="255.255.255.255" description="Broadcast IP from Sensor"/>
    <arg name="data_port" default="2368" description="LiDAR Data Port"/>
    <arg name="gnss_port" default="2369" description="LiDAR GNSS Port"/>

    <arg name="rotation_speed" default="600" description="Motor RPM, the sensor's internal spin rate."/>
    <arg name="cloud_min_angle" default="0" description="Field of View, start degrees."/>
    <arg name="cloud_max_angle" default="360" description="Field of View, end degrees."/>
    <arg name="diag_span" default="1000" description="milliseconds"/>

    <arg name="setup_sensor" default="True" description="Enable sensor setup on hw-driver."/>
    <arg name="retry_hw" default="True" description="hw driver startup retry (false when using pcap)."/>

    <node pkg="nebula_ros" exec="tutorial_ros_wrapper_node"
        name="nebula_tutorial" output="screen">
        <param name="sensor_model" value="$(var sensor_model)"/>
        <param name="return_mode" value="$(var return_mode)"/>
        <param name="frame_id" value="$(var frame_id)"/>
        <param name="scan_phase" value="$(var scan_phase)"/>
        <param name="sensor_ip" value="$(var sensor_ip)"/>
        <param name="calibration_file" value="$(var calibration_file)"/>
        <param name="launch_hw" value="$(var launch_hw)"/>

        <param name="host_ip" value="$(var host_ip)"/>
        <param name="data_port" value="$(var data_port)"/>
        <param name="gnss_port" value="$(var gnss_port)"/>
        <param name="packet_mtu_size" value="$(var packet_mtu_size)"/>
        <param name="rotation_speed" value="$(var rotation_speed)"/>
        <param name="cloud_min_angle" value="$(var cloud_min_angle)"/>
        <param name="cloud_max_angle" value="$(var cloud_max_angle)"/>
        <param name="dual_return_distance_threshold" value="$(var dual_return_distance_threshold)"/>
        <param name="setup_sensor" value="$(var setup_sensor)"/>
        <param name="retry_hw" value="$(var retry_hw)"/>

        <param name="diag_span" value="$(var diag_span)"/>
    </node>
</launch>
